build/ci_utils/src/cache/goodie/graalvm.rs

```diff
@@ -22,24 +22,77 @@ crate::define_env_var! {
     GRAALVM_HOME, PathBuf;
 }
 
-pub fn graal_version_from_version_string(version_string: &str) -> Result<Version> {
-    let line = version_string.lines().find(|line| line.contains("GraalVM CE")).context(
+const CE_JAVA_VENDOR: &str = "GraalVM CE";
+const EE_JAVA_VENDOR: &str = "Oracle GraalVM";
+
+#[derive(Copy, Clone, Debug, PartialEq)]
+pub enum Edition {
+    /// GraalVM CE (Community Edition).
+    Community,
+    /// Oracle GraalVM (Formerly GraalVM EE, Enterprise Edition)
+    Enterprise,
+}
+
+impl std::str::FromStr for Edition {
+    type Err = anyhow::Error;
+
+    fn from_str(s: &str) -> Result<Self> {
+        match s {
+            CE_JAVA_VENDOR => Ok(Edition::Community),
+            EE_JAVA_VENDOR => Ok(Edition::Enterprise),
+            _ => bail!("Unknown GraalVM edition: {}", s),
+        }
+    }
+
+}
+
+impl Into<String> for Edition {
+    fn into(self) -> String {
+        match self {
+            Edition::Community => CE_JAVA_VENDOR.to_string(),
+            Edition::Enterprise => EE_JAVA_VENDOR.to_string(),
+        }
+    }
+}
```

Comment:
In general, if there's `Display` you get `ToString` for free. Usually then `Into<String>` is not defined, especially if it does the same thing.

Reply:
I have "fixed" this by implementing `From<Edition> for String` trait in 66b03bb850fce28411a134a19d74e4060e1746e9, as suggested by the linter.